scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,1075 @@
+        It must be greater than or equal to 0. When set to 0, then sequential
+        search is used. Default is 2.
+    seed : int, BitGenerator, Generator, RandomState, SeedSequence, optional
+        Seed for the underlying uniform random number generation.
+
+    Notes
+    -----
+    The transformation of the PDF must be concave in order to
+    construct the hat function. Such a PDF is called T-concave. Currently
+    the following transformations are supported:
+
+    .. math::
+
+        c = 0.: T(x) &= \log(x)\\
+        c = -0.5: T(x) &= \frac{1}{\sqrt{x}} \text{ (Default)}
+
+    Three variants of this method are available:
+
+    * GW: squeezes between construction points.
+    * PS: squeezes proportional to hat function. (Default)
+    * IA: same as variant PS but uses a compositon method with
+      "immediate acceptance" in the region below the squeeze.
+
+    
+    References
+    ----------
+    .. [1] UNU.RAN reference manual, Section 5.3.16,
+           "TDR - Transformed Density Rejection",
+           http://statmath.wu.ac.at/software/unuran/doc/unuran.html#TDR
+    .. [2] Hörmann, Wolfgang. "A rejection technique for sampling from
+           T-concave distributions." ACM Transactions on Mathematical
+           Software (TOMS) 21.2 (1995): 182-193
+    .. [3] W.R. Gilks and P. Wild (1992). Adaptive rejection sampling for
+           Gibbs sampling, Applied Statistics 41, pp. 337-348.
+
+    Examples
+    --------
+    >>> from scipy.stats import TransformedDensityRejection
+
+    Suppose we have a density:
+
+    .. math::
+
+        f(x) = \begin{cases}
+                1 - x^2,  &  -1 \leq x \leq 1 \\
+                0,        &  \text{otherwise}
+               \end{cases}
+
+    The derivative of this density function is:
+
+    .. math::
+
+        \frac{df(x)}{dx} = \begin{cases}
+                            -2x,  &  -1 \leq x \leq 1 \\
+                            0,    &  \text{otherwise}
+                           \end{cases}
+
+    Notice that the PDF doesn't integrate to 1. As this is a rejection based
+    method, we need not have a normalized PDF. To initialize the generator,
+    we can use:
+
+    >>> urng = np.random.default_rng()
+    >>> class MyDist:
+    ...     def pdf(self, x):
+    ...         return 1-x*x
+    ...     def dpdf(self, x):
+    ...         return -2*x
+    ... 
+    >>> dist = MyDist()
+    >>> rng = TransformedDensityRejection(dist, domain=(-1, 1), seed=urng)
+
+    Now, we can use the `rvs` method to generate samples from the
+    distribution:
+
+    >>> rvs = rng.rvs(1000)
+
+    To use the log transformation with 10 construction points and the
+    immediate acceptance variant of the method, do:
+
+    >>> rng = TransformedDensityRejection(dist, domain=(-1, 1), c=0.,
+    ...                                   cpoints=10, variant='ia', seed=urng)
+    >>> rvs = rng.rvs(1000)
+    """
+    def __cinit__(self,
+                  dist,
+                  mode=None,
```

Comment:
Should we not make the rest keyword only? (not sure if this work in Cython though)
```suggestion
                  dist,
                  *,
                  mode=None,
```

Reply:
> Should we not make the rest keyword only? (not sure if this work in Cython though)

We should and it does work with Cython.

Reply:
Done in (6a6054f14f0c6a6cc978bf6d0785bc0a7d337a53).